src/image/VisitInfo.cc

```diff
@@ -472,11 +472,31 @@ VisitInfo::VisitInfo(daf::base::PropertySet const& metadata)
     }
 }
 
+/**
+ * Test whether two numbers are exactly equal or both NaN.
+ *
+ * This function is exactly equivalent to `operator==(double, double)`, except
+ * that NaNs compare equal (NaNs of different values also compare equal).
+ */
+bool _eqOrNan(double lhs, double rhs) noexcept { return (std::isnan(lhs) && std::isnan(rhs)) || lhs == rhs; }
+
+/**
+ * Test whether two SpherePoints are exactly equal or invalid.
+ *
+ * This function is needed because SpherePoint::operation== is specifically
+ * designed to be ill-behaved when NaN values are involved.
+ */
+bool _eqOrNan(lsst::geom::SpherePoint const& lhs, lsst::geom::SpherePoint const& rhs) noexcept {
+    return (!lhs.isFinite() && !rhs.isFinite()) || lhs == rhs;
+}
+
 bool VisitInfo::operator==(VisitInfo const& other) const {
```

Comment:
Again, same question about whether NaN equality is desirable, yadda yadda yadda.